inference-engine/src/transformations/src/transformations/serialize.cpp

```diff
@@ -151,100 +151,100 @@ class XmlSerializer : public ngraph::AttributeVisitor {
-                    pugi::xml_node iter_input = port_map.append_child("input");
-                    iter_input.append_attribute("external_port_id").set_value("-1");
-                    iter_input.append_attribute("internal_layer_id").set_value(parameter_mapping[a->get().current_iteration_input_idx].c_str());
-                    iter_input.append_attribute("purpose").set_value("current_iteration");
-                }
+    void output_descriptions_on_adapter(const std::vector<std::shared_ptr<
+                                        ngraph::op::util::SubGraphOp::OutputDescription>>& output_descriptions) {
+        std::vector<std::string> result_mapping = map_type_from_body(m_xml_node.parent(), "Result");
+        std::vector<std::string> parameter_mapping = map_type_from_body(m_xml_node.parent(), "Parameter");
 
-                if (a->get().body_condition_output_idx != -1) {
-                    pugi::xml_node exec_output = port_map.append_child("output");
-                    exec_output.append_attribute("external_port_id").set_value("-1");
-                    exec_output.append_attribute("internal_layer_id").set_value(result_mapping[a->get().body_condition_output_idx].c_str());
-                    exec_output.append_attribute("purpose").set_value("execution_condition");
-                }
+        NGRAPH_CHECK(!result_mapping.empty(), "No results found in body Function.");
+
+        pugi::xml_node port_map = m_xml_node.parent().child("port_map");
+        if (!port_map) {
+            port_map = m_xml_node.parent().insert_child_before("port_map", m_xml_node.parent().first_child());
+        }
+
+        for (const auto& output_description : output_descriptions) {
+            pugi::xml_node output = port_map.append_child("output");
+            output.append_attribute("external_port_id").set_value(parameter_mapping.size() + output_description->m_output_index);
+            output.append_attribute("internal_layer_id").set_value(result_mapping[output_description->m_body_value_index].c_str());
+
+            if (auto concat_output = as_type_ptr<ngraph::op::util::SubGraphOp::ConcatOutputDescription>(output_description)) {
+                output.prepend_attribute("axis").set_value(concat_output->m_axis);
+                output.append_attribute("start").set_value(concat_output->m_start);
+                output.append_attribute("end").set_value(concat_output->m_end);
+                output.append_attribute("stride").set_value(concat_output->m_stride);
+                output.append_attribute("part_size").set_value(concat_output->m_part_size);
             }
+        }
+    }
+
+    void special_body_ports_on_adapter(const ngraph::op::v5::Loop::SpecialBodyPorts& special_body_ports) {
+        std::vector<std::string> result_mapping = map_type_from_body(m_xml_node.parent(), "Result");
+        std::vector<std::string> parameter_mapping = map_type_from_body(m_xml_node.parent(), "Parameter");
```

Comment:
`result_mapping` & `parameter_mapping` should be computed in `on_adapter<void>` as before and passed as a reference to new helper functions, now this code is duplicated unnecessarily